File: backend/internal/dto/classroom_invitation_dto.go
package dto

import "time"

// ===== REQUEST DTOs =====

type InviteToClassroomRequest struct {
	Email string `json:"email" binding:"required,email"`
}

// ===== RESPONSE DTOs =====

type ClassroomInvitationResponse struct {
	ID            string    `json:"id"`
	ClassroomID   string    `json:"classroom_id"`
	ClassroomName string    `json:"classroom_name"`
	InviterID     string    `json:"inviter_id"`
	InviterName   string    `json:"inviter_name"`
	InviteeID     string    `json:"invitee_id"`
	InviteeName   string    `json:"invitee_name"`
	InviteeEmail  string    `json:"invitee_email"`
	Status        string    `json:"status"`
	CreatedAt     time.Time `json:"created_at"`
}

File: frontend/src/dtos/classroom-invitation-dto.ts
import type { InvitationStatus } from "../models";

// ==================== REQUEST DTOs ====================

export interface InviteRequest {
  emails: string[];
}

// ==================== RESPONSE DTOs ====================

export interface InviteResult {
  invited: string[];
  alreadyMembers: string[];
  failed: InviteError[];
  totalProcessed: number;
}

export interface InviteError {
  row?: number; // For Excel import
  email: string;
  error: string;
}

export interface InvitationResponse {
  id: string;
  email: string;
  classroomId: string;
  classroomName?: string;
  invitedBy: string;
  status: InvitationStatus;
  expiresAt: string;
  createdAt: string;
}

export interface AcceptInvitationResponse {
  message: string;
  classroomId: string;
}
